fuel_gauge: Add fuel_guage_get_prop/s() (#62827)
zephyrproject-rtos/zephyr: doc/hardware/peripherals/fuel_gauge.rst

Fuel gauges typically support multiple properties, such as temperature readings of the battery-pack
or present-time current/voltage.

> Properties are fetched one at a time using a client allocated :c:struct:`fuel_gauge_property` or
> fetched all at once using a client allocated array of :c:struct:`fuel_gauge_property`. Each
> :c:struct:`fuel_gauge_property` is populated by values according to its `property_type` field.

Reviewer: I find it clearer to tell the user the function call to make.

```suggestion
Properties are fetched by the client one at a time using :c:func:`fuel_gauge_get_prop`, or fetched in a batch using :c:func:`fuel_gauge_get_props`.  These functions fill in the :c:struct:`fuel_gauge_property` allocated by the client according to its `property_type` field.
```

Author: Whoops. Missed this.

Author: I corrected the documentation based on the final changes in the last commit. Is that acceptable?
